docs/user-guide/ingress-path-matching.md

```diff
@@ -0,0 +1,148 @@
+
+### Example
+
+Let the following two ingress definitions be created:
+
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: test-ingress-1
+spec:
+  host: test.com
+  rules:
+  - http:
+      paths:
+      - path: /foo/bar
+        backend:
+          serviceName: test
+          servicePort: 80
+      - path: /foo/bar/
+        backend:
+          serviceName: test
+          servicePort: 80
+```
+
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: test-ingress-2
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /
+spec:
+  host: test.com
+  rules:
+  - http:
+      paths:
+      - path: /foo/bar/.+
+        backend:
+          serviceName: test
+          servicePort: 80
+```
+
+
+
+The ingress controller would define the following location blocks (in this order) within the NGINX template for the `test.com` server: 
+
+```
+location ~* ^/foo/bar/.+\/?(?<baseuri>.*) {
+  ...
+}
+
+location ~* ^/foo/bar/ {
+  ...
+}
+
+location ~* ^/foo/bar {
+  ...
+}
+```
+The following request URI's would match the corresponding location blocks:
+- `test.com/blog/topics/announcements` matches `~* ^/blog/topics/.+\/?(?<baseuri>.*)`
+- `test.com/blog/topics/` matches `^~ /blog/topics/`
+- `test.com/blog/topics` matches `^~ /blog/topics`
+
+__IMPORTANT NOTES__: 
+- paths created under the `rewrite-ingress` are sorted before `\/?(?<baseuri>.*)` is appended. For example if the path defined within `test-ingress-2` was `/foo/.+` then the location block for `^/foo/.+\/?(?<baseuri>.*)` would be the LAST block listed.
+- If the `use-regex` OR `rewrite-target` annotation is used on any Ingress for a given host, then the case insensitive regular expression [location modifier](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) will be enforced on ALL paths for a given host regardless of what Ingress they are defined on.  
+
+
+## Path Priority Edge case
+The exception to this the longest match policy will occur when long regular expressions are used.
+
+### Example
+
+Let the following ingress be defined:
+
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: test-ingress-1
+  annotations:
+    nginx.ingress.kubernetes.io/use-regex: true
+spec:
+  host: test.com
+  rules:
+  - http:
+      paths:
+      - path: /foo/bar/
+        backend:
+          serviceName: test
+          servicePort: 80
+      - path: /foo/bar/[A-Z0-9]{32}
+        backend:
+          serviceName: test
+          servicePort: 80
+```
+
+The ingress controller would define the following location blocks (in this order) within the NGINX template for the `test.com` server: 
+
+```
+location ~* ^/foo/bar/[A-Z0-9]{32} {
+  ...
+}
+
+location ~* ^/foo/bar/ {
+  ...
+}
+```
+
+A request to `test.com/foo/bar` would match the `^/foo/[A-Z0-9]{32}` location block. 
```

Comment:
Is there an explanation why this is the case? An official link by Nginx talking about this or something?